manifests/candlepin.pp

```diff
@@ -47,7 +47,7 @@
     keystore_password            => $certs::candlepin::keystore_password,
     truststore_file              => $certs::candlepin::truststore,
     truststore_password          => $certs::candlepin::truststore_password,
-    artemis_client_dn            => $certs::candlepin::artemis_client_dn,
+    artemis_client_dn            => katello::certificate_subject($certs::candlepin::client_cert),
```

Comment:
I'm not sure if this will work in a client-server mode. Functions run server side, but the certificates live client side. I think you need to use a [deferred function](https://puppet.com/docs/puppet/7.5/deferring_functions.html). I think they're relatively recent (we may need to bump our lower version bound) and initial implementations had some issues I've heard. No personal experience with them.

Reply:
Looks like that would work, from some early testing I would have to update puppet-candlepin so that this parameter accepts a Deferred:

```
Class[Candlepin]: parameter 'artemis_client_dn' expects a String value, got Deferred
```